cmd/kubebuilder/docs/gen.go

```diff
@@ -103,5 +108,8 @@ resource_categories: {{ range $category := .Categories }}
   resources: {{ range $resource := $category.Resources }}
   - name: "{{ $resource.Kind }}"
     version: "{{ $resource.Version }}"
-    group: "{{ $resource.Group }}"{{ end }}
+    group: "{{ $resource.Group }}"{{ if $resource.Warning }}
+    description_warning: "{{ $resource.Warning }}"{{ end -}}{{ if $resource.Note }}
+    description_note: "{{ $resource.Note }}"{{ end -}}
```

Comment:
these lines look a little bit weird.. can you explain?

Reply:
The generated config.yaml is not supported to be modified directly by user, so annotations like `// +kubebuilder:doc:warning` and `// +kubebuilder:doc:note` need to be supported.
As seen in
https://github.com/kubernetes-incubator/reference-docs/blob/master/gen-apidocs/generators/config.yaml

Reply:
sorry i didn't make it clear. I mean the template looks weird. what you want is probably like this
```golang
{{- if $resource.Warning }}
  description_warning: "{{ $resource.Warning }}"
{{- end }}
```

Reply:
Thank you very much! I will make a change in a separate PR.